### ISSUE TITLE
MD020 triggers on escaped hash

### ISSUE THREAD

**Issue by akoeplinger** [markdownlint/markdownlint#56]
A header like the following still triggers MD020 - No space inside hashes on closed atx style header:

```
## Benefits of C\#
...
```

The hash char at the end is escaped, so the rule shouldn't trigger.
